web_transport/sdk/impl/web_transport_owt_client_impl.cc

```diff
@@ -0,0 +1,252 @@
+  base::WaitableEvent done(base::WaitableEvent::ResetPolicy::AUTOMATIC,
+                           base::WaitableEvent::InitialState::NOT_SIGNALED);
+  WebTransportStreamInterface* stream(nullptr);
+  task_runner_->PostTask(
+      FROM_HERE, base::BindOnce(
+                     [](WebTransportOwtClientImpl* client,
+                        WebTransportStreamInterface** result,
+                        bool bidirectional, base::WaitableEvent* event) {
+                       CHECK(client);
+                       *result = client->CreateOutgoingStreamOnCurrentThread(
+                           bidirectional);
+                       event->Signal();
+                     },
+                     base::Unretained(this), base::Unretained(&stream),
+                     bidirectional, base::Unretained(&done)));
+  done.Wait();
+  return stream;
+}
+
+WebTransportStreamInterface*
+WebTransportOwtClientImpl::CreateOutgoingStreamOnCurrentThread(
+    bool bidirectional) {
+  ::quic::WebTransportStream* stream(nullptr);
+  if (bidirectional) {
+    if (!client_->session()->CanOpenNextOutgoingBidirectionalStream()) {
+      return nullptr;
+    }
+    stream = client_->session()->OpenOutgoingBidirectionalStream();
+  } else {
+    if (!client_->session()->CanOpenNextOutgoingUnidirectionalStream()) {
+      return nullptr;
+    }
+    stream = client_->session()->OpenOutgoingUnidirectionalStream();
+  }
+  DCHECK(stream);
+  return OwtStreamForNativeStream(stream);
+}
+
+void WebTransportOwtClientImpl::OnIncomingBidirectionalStreamAvailable() {
+  event_runner_->PostTask(
+      FROM_HERE,
+      base::BindOnce(&WebTransportOwtClientImpl::OnIncomingStreamAvailable,
+                     weak_factory_.GetWeakPtr(), true));
+}
+
+void WebTransportOwtClientImpl::OnIncomingUnidirectionalStreamAvailable() {
+  event_runner_->PostTask(
+      FROM_HERE,
+      base::BindOnce(&WebTransportOwtClientImpl::OnIncomingStreamAvailable,
+                     weak_factory_.GetWeakPtr(), false));
+}
+
+void WebTransportOwtClientImpl::OnIncomingStreamAvailable(bool bidirectional) {
+  auto* stream = bidirectional
+                     ? client_->session()->AcceptIncomingBidirectionalStream()
+                     : client_->session()->AcceptIncomingUnidirectionalStream();
+  CHECK(stream);
+  if (visitor_) {
+    auto* owt_stream = OwtStreamForNativeStream(stream);
+    CHECK(owt_stream);
+    visitor_->OnIncomingStream(owt_stream);
+  } else {
+    // No one cares about incoming streams.
+    DCHECK(stream->SendFin());
+  }
+}
+
+WebTransportStreamInterface*
+WebTransportOwtClientImpl::OwtStreamForNativeStream(
+    ::quic::WebTransportStream* stream) {
+  std::unique_ptr<WebTransportStreamImpl> stream_impl =
+      std::make_unique<WebTransportStreamImpl>(stream, task_runner_.get(),
+                                               event_runner_.get());
+  WebTransportStreamImpl* stream_ptr(stream_impl.get());
+  streams_.push_back(std::move(stream_impl));
```

Comment:
where do you pop this stream from streams_?

Reply:
It should be poped when the stream is closed, but it's not implemented.